liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -957,6 +976,25 @@ public boolean recompileBuildFile(File buildFile, Set<String> compileArtifactPat
                     }
 
                 }
+                
+                // We don't currently have the ability to dynamically add new directories to be watched
+                // There is so much that we are dynamically able to do that this could be surprising.
+                // For now issue a warning
+                Set<Path> oldMonitoredWebResourceDirs = new HashSet<Path>(this.monitoredWebResourceDirs);
+                Set<Path> newMonitoredWebResourceDirs = new HashSet<Path>(LooseWarApplication.getWebSourceDirectoriesToMonitor(project));
+                if (!oldMonitoredWebResourceDirs.equals(newMonitoredWebResourceDirs)) {
+                    getLog().warn("Change detected in the set of filtered web resource directories, since dev mode was first launched.  Adding/deleting a web resource directory has no change on the set of directories monitored by dev mode.  Changing the watch list will require a dev mode restart");
+                }
+                
+                // convert to Path, which seems to offer more reliable cross-platform, relative path comparison, then compare
+                Set<Path> oldResourceDirs = new HashSet<Path>();
+                this.resourceDirs.forEach(r -> oldResourceDirs.add(r.toPath()));                
```

Comment:
This seems to be using `toPath()` on File objects, which in the common PR you explicitly called `getCanonicalFile().toPath()` on a File object. Wondering if you need to do the same here?

Reply:
It is fine if you leave it. Just wondering the difference.